pages/utilities/css-units-converter.tsx

```diff
@@ -0,0 +1,207 @@
+        convertedValue = conversionFunction(value, containerWidthValue);
+      } else if (["px-vh", "vh-px"].includes(conversionKey)) {
+        convertedValue = conversionFunction(value, containerHeightValue);
+      } else if (["px-vmin", "vmin-px"].includes(conversionKey)) {
+        convertedValue = conversionFunction(
+          value,
+          containerWidthValue,
+          containerHeightValue
+        );
+      } else if (["px-vmax", "vmax-px"].includes(conversionKey)) {
+        convertedValue = conversionFunction(
+          value,
+          containerWidthValue,
+          containerHeightValue
+        );
+      } else {
+        convertedValue = conversionFunction(value);
+      }
+
+      setOutput(convertedValue.toString());
+    }
+  }, [
+    fromUnit,
+    heightInput,
+    htmlInput,
+    needsContainerInput,
+    toUnit,
+    widthInput,
+  ]);
+
+  return (
+    <main>
+      <Meta
+        title="CSS units converter by Jam.dev | Free, Open Source & Ad-free"
+        description="Convert your css units values to other options"
+      />
+      <Header />
+      <CMDK />
+      <section className="container max-w-2xl mb-12">
+        <PageHeader
+          title="CSS Units Converter"
+          description="Convert your css units values to other options"
+        />
+      </section>
+      <section className="container max-w-2xl mb-6">
+        <Card className="flex flex-col p-6 hover:shadow-none shadow-none rounded-xl">
+          <div>
+            <Label>Input Value</Label>
+            <Textarea
+              className="mb-6"
+              rows={3}
+              value={htmlInput}
+              onChange={(e) => setHtmlInput(e.target.value)}
+            />
+            <div className="mb-6 flex w-full gap-4">
+              <div className="flex flex-1 flex-col">
+                <Label>From Unit</Label>
+                <Combobox
+                  data={unitOptions}
+                  onSelect={(value) => setFromUnit(value)}
+                />
+              </div>
+              <div className="flex flex-1 flex-col">
+                <Label>To Unit</Label>
+                <Combobox
+                  data={unitOptions}
+                  onSelect={(value) => setToUnit(value)}
+                />
+              </div>
+            </div>
+            {needsContainerInput && (
+              <div className="mb-6">
+                <Label>Container Width (px)</Label>
+                <Textarea
+                  className="mb-6"
+                  rows={1}
+                  value={widthInput}
+                  onChange={(e) => setWidthInput(e.target.value)}
+                />
+                <Label>Container Height (px)</Label>
+                <Textarea
+                  className="mb-6"
+                  rows={1}
+                  value={heightInput}
+                  onChange={(e) => setHeightInput(e.target.value)}
+                />
+              </div>
+            )}
```

Comment:
It might be beneficial to use `Input` instead of `Textarea` and have 50/50 layout.
With this approach UI is more compact and feels easier to digest.

Code suggestion:

```
{needsContainerInput && (
  <div className="mb-6 flex gap-4">
    <div className="flex flex-1 flex-col">
      <Label>Container Width (px)</Label>
      <Input
        type="number"
        className="mb-6 h-[32px]"
        value={widthInput}
        onChange={(e) => setWidthInput(e.target.value)}
      />
    </div>
    <div className="flex flex-1 flex-col">
      <Label>Container Height (px)</Label>
      <Input
        type="number"
        className="mb-6 h-[32px]"
        value={heightInput}
        onChange={(e) => setHeightInput(e.target.value)}
      />
    </div>
  </div>
)}
```

---

<img width="641" alt="image" src="https://github.com/user-attachments/assets/0d0d3944-b3ee-4ebe-bf0e-07d35d583e4c">